### ISSUE TITLE
API SAMI Add canonical urls to all pages

### ISSUE THREAD

**Issue by tractorcow** [silverstripe/api.silverstripe.org#47]
After the migration to sami there was an issue adding canonical urls;

This is the code I was using
```twig
<link rel="canonical"
    href="{{ project.config('base_url')|replace({'%version%': project.version.name}) }}/{{ uri }}"
/>
```

Since documents are cached between versions (e.g. 4, master, 3) it's not possible to render version-specific canonical urls, since it pollutes other versions.

I've tracked the code which pollutes the cache to `Project::renderVersion`

```
if ($previous && !$this->renderer->isRendered($this)) {
    $this->seedCache($previous, $this->getBuildDir());
}
```

However, removing this line causes so much extra work the task runs out of memory and crashes, so it needs to be left in for now.

**Comment by williamdes** [comment]
You could open an issue at https://github.com/code-lts/doctum ;)

**Comment by chillu** [comment]
I'm not quite sure why we needed canonical in the metadata in the first place.
https://api.silverstripe.org/4/SilverStripe/Admin/Forms.html is different from https://api.silverstripe.org/master/SilverStripe/Admin/Forms.html, and both those representations only exist once per version.

I've had a quick look at [Google search results](https://www.google.com/search?client=safari&source=hp&ei=933aX_SwFsCR4-EP3qyOgAY&q=dataobject+site%3Aapi.silverstripe.org&oq=dataobject+site%3Aapi.silverstripe.org&gs_lcp=CgZwc3ktYWIQAzoRCC4QsQMQxwEQowIQyQMQkwI6CAgAELEDEIMBOg4ILhCxAxCDARDHARCjAjoFCAAQsQM6AggAOgUILhCxAzoOCC4QsQMQgwEQxwEQrwE6AgguOggILhCxAxCDAToLCAAQsQMQgwEQyQM6CAguEMcBEK8BOgUIABDJAzoKCAAQsQMQgwEQCjoECAAQCjoKCC4QxwEQrwEQCjoGCAAQFhAeOgUIIRCgAVCICljJLGCfLWgBcAB4AYABiQKIAcAzkgEGMC4yOC43mAEAoAEBqgEHZ3dzLXdpeg&sclient=psy-ab&ved=0ahUKEwi0iN_kutPtAhXAyDgGHV6WA2AQ4dUDCAo&uact=5) and they seem to be handling this fine (pointing to `/4/*` by default). Presumably Google takes the signal of default redirections and makes it influence the search results?

Either way, we've migrated to Doctum now, so the Sami based comments no longer apply. Closing.
